src/session/content/chat_action_bar.rs

```diff
@@ -554,7 +621,71 @@ impl ChatActionBar {
                 "label",
                 Some(chat),
             );
-            bindings.push(restriction_label_binding)
+            bindings.push(restriction_label_binding);
+
+            // Handle [un]mute button visibility
+            let mute_btn_visibility = gtk::ClosureExpression::with_callback(&[Chat::this_expression("type")], |args| {
+            matches!(args[1].get::<ChatType>().unwrap(), ChatType::Supergroup(data) if data.is_channel() && !matches!(data.status().0, ChatMemberStatus::Administrator(_) | ChatMemberStatus::Creator(_) | ChatMemberStatus::Left))
+        })
+            .upcast()
+            .bind(&*imp.mute_button, "visible", Some(chat));
```

Comment:
All this bindings can probably be simplified by not using expressions, especially if you use a GtkStack. An example of how this can be reworked:

- Create a function to update the stack page based on the chat object properties.
- Connect notify signals from chat properties and make them call that update function.

Reply:
> Frankly speaking, I do not have any idea how to do this

Yeah it's a bit complex, actually I'd like to use a new glib API to simplify this a bit. If you push the changes to fix the other reviews I can do it myself after this is merged, if you're ok with that.